### DISCUSSION TITLE
await usefetch does not work when called in onMounted. (SSR NUXT 3.0.0)

### DISCUSSION THREAD

**Discussion by SalmaBelal** [nuxt/nuxt#18675] [General]
When i call usefetch with an await inside onMounted, it doesnt wait for the response to return. it returns in pending state.

**Comment by Baroshem** [comment]
`useFetch` should not be used inside `onMounted`.

You should be using just `await useFetch` as it is described here 

https://nuxt.com/docs/getting-started/data-fetching#example

**Reply by SalmaBelal** [reply]
But the docs say "useFetch, useLazyFetch, useAsyncData and useLazyAsyncData only work during setup or Lifecycle Hooks". Onmounted is a lifecycle hook.

**Reply by TheAlexLichter** [reply]
I think we should improve the docs there for sure (@danielroe we briefly talked about it last meeting).

Best in this case would be to call `useFetch` with `server: false` I'd say.
